spark/v3.5/spark/src/main/java/org/apache/iceberg/spark/procedures/BaseProcedure.java

```diff
@@ -57,6 +58,14 @@ abstract class BaseProcedure implements Procedure {
       DataTypes.createMapType(DataTypes.StringType, DataTypes.StringType);
   protected static final DataType STRING_ARRAY = DataTypes.createArrayType(DataTypes.StringType);
 
+  protected static ProcedureParameter requiredInParameter(String name, DataType dataType) {
+    return ProcedureParameter.required(name, dataType);
+  }
+
+  protected static ProcedureParameter optionalInParameter(String name, DataType dataType) {
+    return ProcedureParameter.optional(name, dataType);
+  }
```

Comment:
any reason why we didn't have this in 4.0 ? are we missing backport of some commit to 3.5 
seems like not - https://github.com/apache/iceberg/commit/30ee7e83cfa3e5559d8d3ba4b6380156accb43f1 

can you please double confirm ?

Reply:
Thank you very much for reviewing the code! This is not a clean backport, since there is a PR (#13106) in the Spark 4.0 branch that cannot be directly backported to Spark 3.4/3.5. Because Spark 3.5 lacks some features compared to Spark 4.0, I made certain adjustments based on the current Spark 3.5 code.

Reply:
In Spark 3.5, we use `org.apache.spark.sql.connector.iceberg.catalog.ProcedureParameter` for parameter parsing, whereas in the Spark 4 branch, `org.apache.spark.sql.connector.catalog.procedures.ProcedureParameter` is used to achieve the same functionality. Personally, I believe the difference in functionality between the two is not significant.